### PR TITLE
Get rid of CuffSystem, make CuffableComponent update its hand count properly

### PR DESCRIPTION
**Pull request by nuke-makes-games**

### CODE REVIEW

**Review by Zumorica — APPROVED** [review]

File: Content.Server/GameObjects/Components/GUI/HandsComponent.cs [hunk 1]
```diff
@@ -467,6 +468,11 @@ public void RemoveHand(string name)
 
             OnItemChanged?.Invoke();
 
+            if (Owner.TryGetComponent<CuffableComponent>(out var cuffable))
```
Comment (RemieRichards) [inline Content.Server/GameObjects/Components/GUI/HandsComponent.cs:471]: 😬 big yikes on this

Should probably just be a component message, that `CuffableComponent` can then listen to.

Check out the `EventBus`.

**Review by RemieRichards — CHANGES_REQUESTED** [review]
Use a component message (see other comment)

File: Content.Server/GameObjects/Components/GUI/HandsComponent.cs (hunk 1 above)
Comment (RemieRichards) [reply Content.Server/GameObjects/Components/GUI/HandsComponent.cs:471]: Something like:
```cs
    public class HandCountChangedEvent : EntitySystemMessage
    {
        public HandCountChangedEvent()
        {
        }
    }
```

You'd have this here (the line of this comment) in `RemoveHand()`:
```cs
Owner.EntityManager.EventBus.RaiseEvent(EventSource.Local, new HandCountChangedEvent());
```

And anything that cares about this (e.g. `CuffableComponent`) would do:
```cs
Owner.EntityManager.EventBus.SubscribeEvent<HandCountChangedEvent>(EventSource.Local, this, HandleHandCountChange);

...

private void HandleHandCountChange(HandCountChangedEventmessage)
{
	if (message.Entity != Owner)
	    return;
        ... //Do something now you know about the change
}

```

**Review by RemieRichards — APPROVED** [review]
Nice!

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**